### PR TITLE
refactor: redux refactor to use redux toolkit

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -** 
Updated redux action creation to use redux-toolkit and update associated reducer/selector/component usage accordingly
Split out thunkActions from actions
Add/update tests for actions, thunkActions, selectors, and reducers.
Add some docstrings for selectors. 

JIRA:
 [JIRA-5780](https://openedx.atlassian.net/browse/JIRA-5780)
 [JIRA-5781](https://openedx.atlassian.net/browse/JIRA-5781)
 [JIRA-5801](https://openedx.atlassian.net/browse/JIRA-5801)

**Developer Checklist**
- [x] Test suites passing
- [x] Documentation and test plan updated, if applicable
- [x] Received code-owner approving review
- [x] Bumped version number [package.json](../package.json)

**Testing Instructions**

[ How should a reviewer test this PR? ]

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [x] I've done a visual code review
- [x] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: src/data/selectors/cohorts.js [hunk 1]
```diff
@@ -1,16 +1,34 @@
...
+/**
+ * getCohortById(state, selectedCohortId)
+ * returns cohort with given id
+ * @param {object} state - redux state
+ * @param {number} selectedCohortId - id of cohort to return
+ * @return {object} cohort with given id.
+ */
 const getCohortById = (state, selectedCohortId) => {
-  const cohort = allCohorts(state).find(coh => coh.id === selectedCohortId);
+  const cohort = allCohorts(state).find(({ id }) => id === selectedCohortId);
```
Comment (nsprenkle) [inline src/data/selectors/cohorts.js:18]: This makes me irrationally happy when compared to the original

File: src/data/thunkActions/grades.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  formatMinAssignmentGrade,
+  formatMaxCourseGrade,
+  formatMinCourseGrade,
+  formatGradeOverrideForDisplay,
+} = selectors.grades;
+
+export const defaultAssignmentFilter = 'All';
+
+export const fetchBulkUpgradeHistory = courseId => (
+  // todo add loading effect
```
Comment (nsprenkle) [inline src/data/thunkActions/grades.js:26]: Is this todo actionable now or a future todo?

File: src/data/actions/grades.test.js [hunk 3]
```diff
@@ -1,274 +1,115 @@
...
+      test('finished action', () => testAction(actions.fetching.finished));
+      test('error action', () => testAction(actions.fetching.error));
+      describe('received', () => {
+        it('loads grades data from data', () => {
+          const data = {
+            grades: ['some', 'grades'],
+            cohort: 2,
+            track: 'summoners',
+            assignmentType: 'potion',
+            headings: ['H', 'E', 'a', 'd', 'Ing', 'sssss'],
```
Comment (nsprenkle) [inline src/data/actions/grades.test.js:62]: All of your test data sounds ghostly 👻

**Review by nsprenkle — COMMENTED** [review]
Did a cursory read through, generally looks good. Will smoke test.

File: src/data/actions/cohorts.js [hunk 4]
```diff
@@ -1,31 +1,15 @@
...
-  }
-);
-
-export {
-  fetchCohorts,
-  startedFetchingCohorts,
-  gotCohorts,
-  errorFetchingCohorts,
+const fetching = {
+  started: createAction('startedFetching'),
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply src/data/actions/cohorts.js:8]: not a good one.

File: src/data/actions/config.test.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import actions, { dataKey } from './config';
+import { testAction, testActionTypes } from './testUtils';
+
+describe('actions.cohorts', () => {
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply src/data/actions/config.test.js:4]: Ehhh... kind of, but there is value in doing it in a per-module basis

testing of redux modules (aside from selectors), should be suuuuper formulaic to the point of being able to be set up w/ test utils, which is what I am starting on here.

File: src/data/actions/filters.js [hunk 6]
```diff
@@ -35,47 +27,19 @@ const initializeFilters = ({
     courseGradeMax,
     includeCourseRoleMembers: Boolean(includeCourseRoleMembers),
   },
+}));
+
+const reset = createAction('reset'); // paylaod
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply src/data/actions/filters.js:32]: oh, just kind of noting which ones were intended to be called with payload values.   Not sure I'm 100% consistent about it, so I should probably pull these.

File: src/data/actions/roles.js [hunk 7]
```diff
@@ -1,46 +1,14 @@
...
+import { StrictDict } from 'utils';
+import { createActionFactory } from './utils';
 
-const { allFilters } = filtersSelectors;
+export const dataKey = 'roles';
+const createAction = createActionFactory(dataKey);
 
-const allowedRoles = ['staff', 'instructor', 'support'];
+const fetching = {
+  error: createAction('errorFetching'),
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply src/data/actions/roles.js:8]: similar shrug of agreement.

File: src/data/thunkActions/grades.js (hunk 2 above)
Comment (muselesscreator) [reply src/data/thunkActions/grades.js:26]: I would argue out of scope.  Not sure the goal for it is described anywhere.

Comment (nsprenkle) [reply src/data/thunkActions/grades.js:26]: Oh, like "add a loading spinner"? I wasn't sure if this was a "I am requesting we add new stuff" or a "I haven't done this yet but need to".

It it's an "I am requesting we add new stuff", then I'm okay to leave it out of scope.

**Review by nsprenkle — APPROVED** [review]
I've done a smoke test locally on everything but interventions report, which I suggest we test in sandbox before rollout.
